File: 206-ReverseLinkedList/206-ReverseLinkedList.py
# Last updated: 7/31/2026, 8:28:43 PM
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        previous = None
        current = head
        while current:
            forward = current.next
            current.next = previous
            previous = current
            current = forward
        return previous

File: 206-ReverseLinkedList/206-ReverseLinkedList.cpp
// Last updated: 7/31/2026, 8:28:49 PM
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* curr = head;
        ListNode* prev = nullptr;
        ListNode* fwd = nullptr;

        while (curr != nullptr) {
            fwd = curr->next;
            curr->next = prev;
            prev = curr;
            curr = fwd;
        }
        return prev;
    }
};